lib/manifest.js

```diff
@@ -0,0 +1,419 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+'use strict';
+
+const archiver = require('archiver');
+const del = require('del');
+const fs = require('fs');
+const globby = require('globby');
+const jsondiffpatch = require('jsondiffpatch').create({
+    arrays: {
+        detectMove: false
+    }
+});
+const path = require('path');
+const sha1 = require( 'node-sha1' );
+
+const packageJson = require('./../package.json');
+
+/**
+ * @type {String}
+ */
+const FILENAME = 'deployment_manifest.json';
+/**
+ * @type {Regexp}
+ */
+const CARTRIDGE_NAME_REGEX = /cartridges\/([a-z_]+)\/cartridge/;
+/**
+ * @type {Array}
+ */
+const DEFAULT_IGNORE_PATTERNS = [
+    'node_modules/**/*',
```

Comment:
@jordanebachelet  some customer are pulling sfra as a depenendcy through package json. 
this will cause sfra to be stored in node_modules.
Will this still work with that ignore_patterns ?

Reply:
Good point. Let me remove this from the ignore patterns then. Users can still add it back in case they want to ignore the node_modules folder.
Note that the tool only gets the `folder/cartridges/**/*` folders, meaning only folders within a `cartridges` parent folder